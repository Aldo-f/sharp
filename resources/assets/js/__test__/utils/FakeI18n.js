export default {
    install() {
        window.i18n = {};
    }
}