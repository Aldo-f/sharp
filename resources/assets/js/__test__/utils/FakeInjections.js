import Vue from 'vue';

const injectedComponents = ['$field', '$form'];

export default {
    provide() {
        return injectedComponents.reduce((injections, prop) => {
            injections[prop] = new Vue();
            return injections;
        }, {});
    }
}