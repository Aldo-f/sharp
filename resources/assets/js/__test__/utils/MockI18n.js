import Localization ,* as localizationFn from '../../mixins/Localization';

const mockLangImplementation = localeKey => `{{ ${localeKey} }}`;

export default {
    install() {
        window.i18n = {};
    },

    mockLangFunction() {
        localizationFn.lang = jest.fn(mockLangImplementation);
        Localization.methods.l = jest.fn(mockLangImplementation);
    }
}