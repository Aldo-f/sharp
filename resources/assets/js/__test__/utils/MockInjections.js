import Vue from 'vue';

const injectedComponents = {
    $field:{},
    $form:{
        data: () => ({
            errors: {}
        })
    },
    $tab:{}
};

export default {
    provide() {
        return Object.keys(injectedComponents).reduce((injections, compName) => {
            let Comp = Vue.extend(injectedComponents[compName]);
            injections[compName] = new Comp();
            return injections;
        }, {});
    }
}