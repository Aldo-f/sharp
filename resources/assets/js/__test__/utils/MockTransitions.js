export default {
    install() {
        window.getComputedStyle = jest.fn(() => ({
            animationDelay:'',
            animationDuration:'',
            transitionDelay:'',
            transitionDuration:''
        }));
    }
}