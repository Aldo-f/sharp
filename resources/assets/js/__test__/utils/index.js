export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC } from './mockSFC';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}