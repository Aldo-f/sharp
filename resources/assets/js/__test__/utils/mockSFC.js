export function mockSFC (sfcModule, { template, render, ...extend }={}, mockOptions={}) {

    let { isJsFile } = mockOptions;

    if(isJsFile) {
        sfcModule.__oldDefault__ = { ...sfcModule.default }
    }

    Object.assign(sfcModule.default, {
        mixins: [],
        'extends': extend,
        template: template,
        render: !template ? render || function() { return this._v(`__MOCKED_SFC_${sfcModule.name}__\n`) } : null,
        watch: {},
        beforeCreate: ()=>{},
        created: ()=>{},
        beforeMount: ()=>{},
        mounted: ()=>{},
        beforeUpdate: ()=>{},
        updated: ()=>{},
        beforeDestroy: ()=>{},
        destroyed: ()=>{},
        data: ()=>({}),
        provide: {},
        inject: [],
    });

    //console.log(sfcModule);
    return sfcModule.default;
}

export function unmockSFC(sfcModule) {
    if(sfcModule.__oldDefault__) {
        sfcModule.default = sfcModule.__oldDefault__;
        return;
    }

    let { 'default':defaultObj, ...options } = sfcModule;
    sfcModule.default = options;
}