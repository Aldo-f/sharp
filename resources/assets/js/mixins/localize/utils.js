export const LocalizableFields = [
    'text', 'markdown', 'textarea', 'wysiwyg', 'select', 'autocomplete', 'tags'
];
export const LocalizableOptionsFields = [
    'select', 'autocomplete', 'tags'
];
export const LocalizableValueFields = [
    'text', 'textarea', // localized value handled in Form/FieldDisplay components
    //'markdown', 'wysiwyg' (markdown/wysiwyg fields handle localized value themselves)
];

export function isLocaleObject(obj, locales) {
    return obj && typeof obj === 'object' && Object.keys(obj).every(key => locales.includes(key));
}

function isEmpty(value){
    return value === null || value === '';
}
export function isLocaleObjectEmpty(obj) {
    return Object.entries(obj).every(([locale, value])=>isEmpty(value));
}

export function isLocalizableValueField(field) {
    return LocalizableValueFields.includes(field.type);
}

export function localeObject({ locales, resolve=()=>null }) {
    return locales.reduce((res, locale)=>({
        ...res, [locale]: resolve(locale)
    }), {});
}

