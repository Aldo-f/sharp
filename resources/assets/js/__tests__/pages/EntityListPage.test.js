import Vue from 'vue';
import SharpEntityListPage from '../../components/pages/EntityListPage.vue';

import {mockChildrenComponents} from "../utils/mockSFC";
import merge from 'lodash/merge';

// import moxios from 'moxios';
import { MockI18n } from "../utils";
// import { setter, mockProperty, unmockProperty } from "../utils/mock-utils";
// import { nextRequestFulfilled } from '../utils/moxios-utils';
// import HTMLElementsSerializer from '../utils/htmlElementsSnapshotSerializer';
import { shallowMount } from '@vue/test-utils';

jest.mock('../../components/DynamicViewMixin');

describe('EntityListPage', () => {
    Vue.use(MockI18n);
    MockI18n.mockLangFunction();


    function createWrapper({ routeQuery, ...options }={}) {
        const wrapper = shallowMount(SharpEntityListPage, {
            extends: options,
            data() {
                return {
                    ready: true
                }
            },
            stubs: {
                'SharpDataList': `<div class="MOCKED_SharpDataList"> <slot name="item" :item="{}" /> </div>`,
                'SharpDataListRow': `<div class="MOCKED_SharpDataListRow"> <slot name="append" /> </div>`,
                'SharpDropdown':`<div class="MOCKED_SharpDropdown"> <slot /> </div>`,
                'SharpDropdownItem': `<div class="MOCKED_SharpDropdownItem"> <slot /> </div>`,
                'SharpCommandsDropdown': `<div class="MOCKED_SharpCommandsDropdown"> <slot /> </div>`,
                'SharpStateIcon': `<div class="MOCKED_SharpStateIcon"> <slot /> </div>`
            },
            mocks: {
                $route: {
                    params: {
                        id: 'spaceship'
                    },
                    query: {
                        ...routeQuery
                    }
                }
            }
        });
        return wrapper;
    }

    function createData(data) {
        return merge({
            config: {
                commands: {}
            }
        }, data);
    }

    test('can mount entity list', ()=>{
        const wrapper = createWrapper({
            data:() => createData({
                containers: {
                    title: {
                        key: 'title',
                        label: 'Titre'
                    }
                },
                layout: [{
                    key: 'title',
                    size: 6,
                    sizeXS: 12
                }],
                data: {
                    items : [{ id: 1, title: 'Super title' }]
                },
                config: {
                    filters: [],
                    instanceIdAttribute: 'id'
                },
                authorizations:{
                    view: true,
                    update: true
                }
            })
        });

        expect(wrapper.html()).toMatchSnapshot();
    });
});