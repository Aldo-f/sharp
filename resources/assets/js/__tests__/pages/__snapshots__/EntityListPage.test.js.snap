// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityListPage mount 1`] = `
<div class="SharpEntityListPage">
	<sharpactionbarlist-stub count="0" search="" commands=""></sharpactionbarlist-stub>
	<div class="MOCKED_SharpDataList">
		<div url="instanceFormUrl" class="MOCKED_SharpDataListRow"></div>
	</div>
	<sharpcommandformmodal-stub></sharpcommandformmodal-stub>
	<sharpcommandviewpanel-stub></sharpcommandviewpanel-stub>
</div>
`;

exports[`EntityListPage mount with row actions 1`] = `
<div class="SharpEntityListPage">
	<sharpactionbarlist-stub count="0" search="" commands=""></sharpactionbarlist-stub>
	<div class="MOCKED_SharpDataList">
		<div url="instanceFormUrl" class="MOCKED_SharpDataListRow">
			<div class="row">
				<div class="col-auto col-md-12 my-1">
					<div class="MOCKED_SharpDropdown SharpEntityList__state-dropdown">
						<div color="instanceStateIconColor" class="MOCKED_SharpStateIcon"></div>
						<span class="text-truncate">
							instanceStateLabel
						</span>
						<div class="MOCKED_SharpDropdownItem">
							<div color="#000" class="MOCKED_SharpStateIcon"></div>&nbsp; state 1
						</div>
					</div>
				</div>
				<div class="col-auto col-md-12 my-1">
					<div class="MOCKED_SharpCommandsDropdown SharpEntityList__commands-dropdown">
						{{ entity_list.commands.instance.label }}
					</div>
				</div>
			</div>
		</div>
	</div>
	<sharpcommandformmodal-stub></sharpcommandformmodal-stub>
	<sharpcommandviewpanel-stub></sharpcommandviewpanel-stub>
</div>
`;
